metrics/metric.go

```diff
@@ -26,13 +27,17 @@ func InitializeMetrics(defaultCardinality int, cardinalityMap map[string]int) {
 	generators = makeLabelGeneratorMap(defaultCardinality, cardinalityMap)
 }
 
-func NewMetrics(name string, labelCount int) *Metric {
+func NewMetrics(name string, labelCount int, parentLabel string) *Metric {
 	var labels []string
+	metric := new(Metric)
+	if parentLabel != "" {
+		metric.parentName = parentLabel
+		labels = append(labels, parentLabel)
```

Comment:
This will cause for the metric object to initially have randomly generated id for the parentLabel? I see that it actually gets updated with the actual Parent ID in Publish().

Will it cause any issues wrt metrics collected?

Reply:
It should not. Publish is where the metric is actually submitted to prometheus library with a value. We should be good.
"cycle_id" also has been implemented similarly I think